### PR TITLE
deps: Upgrade pre-commit for newer versions of python

### PR DESCRIPTION
**Pull request by delfick**
A remake of https://github.com/typeddjango/django-stubs/pull/1957 as suggested by @sobolevn in https://github.com/typeddjango/django-stubs/issues/1931#issuecomment-1952139286

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Dev tools:
 black==24.2.0
-pre-commit==3.5.0
+pre-commit==3.5.0; python-version < 3.9
+pre-commit==3.6.2
```
Comment (jamesbraza) [inline requirements.txt]: I think you need `; python_version >= 3.9` at the end of this line, otherwise Python 3.8 will get confused as to which to use

Comment (delfick) [reply requirements.txt]: good point, fixed

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Dev tools:
 black==24.2.0
-pre-commit==3.5.0
+pre-commit==3.5.0; python-version < 3.9
+pre-commit==3.6.2; python-version >= 3.9
```
Comment (sobolevn) [inline requirements.txt]:
```suggestion
pre-commit==3.5.0; python_version < 3.9
pre-commit==3.6.2; python_version >= 3.9
```

**Review by sobolevn — APPROVED** [review]
Thanks!

File: requirements.txt (hunk 2 above)
Comment (delfick) [reply requirements.txt]: so close!

**Comment by delfick** [comment]
you're welcome!

**Pull request merged**